src/main/python/pybuilder/plugins/python/coverage_plugin.py

```diff
@@ -130,38 +142,87 @@ def do_coverage(project, logger, reactor, execution_prefix, execution_name, targ
 
         if module not in modules:
             modules.append(module)
+    return modules
+
+
+def _build_module_report(coverage, module):
+    return ModuleCoverageReport(coverage._analyze(module))
```

Comment:
@mriehl just to make sure this isn't skipped: we're now using a semi-internal version of the API. I emailed the maintainer asking to open it up, haven't heard back yet though, but it has been stable since 3.x and is underpinning both `analysis` and `analysis2` methods of `coverage`.